core/src/test/java/org/infinispan/api/SizeOptimizationTests.java

```diff
@@ -128,6 +133,9 @@ public ConfigurationBuilder configure(ConfigurationBuilder builder) {
             builder.persistence()
                   .passivation(false)
                   .clearStores();
+            builder.transaction()
```

Comment:
Not related to this line, but one thing I would suggest is replacing the IDC with another that throws an exception when the iterator or any publish methods are invoked on it for the duration of the test (for tests that should be using the optimization). Then put the original back in before shutting down. This will catch any optimization failures as it shouldn't be iterating over the contents.